### PR TITLE
[FEATURE] Implement Integration Test for File Path Data Connectors with Partitioning, Splitting, and Sampling

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Note
Besides its general utility, this integration test will be also used for `DocuSaurus`-based documentation, covering the subject of data introspection and partition for filesystem based assets.  (Documentation will be the focus of an upcoming pull request.)

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-200/GREAT-221/GREAT-277
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/integration/docusaurus/connecting_to_your_data/how_to_introspect_and_partition_your_data/files/yaml_example.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
+import os
+
+from ruamel import yaml
+
+import great_expectations as ge
+from great_expectations.core.batch import BatchRequest
+
+context = ge.get_context()
+data_dir_path = os.path.join("..", "data")
```
Comment (cdkini) [inline tests/integration/docusaurus/connecting_to_your_data/how_to_introspect_and_partition_your_data/files/yaml_example.py:9]: Should this be `os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..', 'data'))`?

Relative pathing can be weird sometimes so I'd rather be explicit and use abspath if possible.

**Review by cdkini — COMMENTED** [review]
Instead of duplicating these 2020 files, could we possibly write a helper function or pytest fixture to set up `sample_2020` in a tmpdir? I understand the need to separate out the data for purposes of this test here but I'd rather we not duplicate 120K lines of data if possible.

Also left a minor comment regarding pathing.

Let me know your thoughts!

File: tests/integration/docusaurus/connecting_to_your_data/how_to_introspect_and_partition_your_data/files/yaml_example.py (hunk 1 above)
Comment (alexsherstinsky) [reply tests/integration/docusaurus/connecting_to_your_data/how_to_introspect_and_partition_your_data/files/yaml_example.py:9]: @cdkini I am following the pattern of the integration tests, which are enabled by `tests/integration/test_script_runner.py` -- there are provisions to set up the `data` directory in the way that it is referenced.  If we are to change this, then I would propose it to be a separate effort.  Thanks! /cc @Shinnnyshinshin

**Comment by alexsherstinsky** [comment]
> Instead of duplicating these 2020 files, could we possibly write a helper function or pytest fixture to set up `sample_2020` in a tmpdir? I understand the need to separate out the data for purposes of this test here but I'd rather we not duplicate 120K lines of data if possible.
> 
> Also left a minor comment regarding pathing.
> 
> Let me know your thoughts!

@cdkini I agree in principle.  However, building this helper function would take some thought and time.  So perhaps we can address it as a separate effort (e.g., as part of a code cleanup and/or a Hackathon project)?  Thanks! /cc @Shinnnyshinshin

**Review by cdkini — APPROVED** [review]
As long as we've got a ticket to clean up fixtures in the backlog, I'm happy to approve this.

Thanks for the hard work!

**Comment by alexsherstinsky** [comment]
> As long as we've got a ticket to clean up fixtures in the backlog, I'm happy to approve this.
> 
> Thanks for the hard work!

The ticket has been created and added to the backlog.  Thank you!
